src/main.ts

```diff
@@ -280,15 +285,19 @@ export default class RecipeGrabber extends Plugin {
 			}
 
 			// pages can have multiple recipes, lets add them all
-			let i = 0;
 			for (const recipe of recipes) {
 				if (this.settings.debug) {
 					console.log(recipe);
 					console.log(markdown(recipe));
 				}
 				// this will download the images and replace the json "recipe.image" value with the path of the image file.
 				if (this.settings.saveImg && file) {
-					const filename = recipe.name.replace(/\ /g, "-"); // We dont want spaces in filenames
```

Comment:
had some type issues here, also, current is replacing all spaces with `-`. thinking you wanted any consecutive spaces with a single dash.

Reply:
Yep. We could easily pull in a library to do this right, but i think what you have is fine for most use cases.

Reply:
My regex change only really adds multiple spaces with a single `-`